### PR TITLE
config: move config.rs files to src/config

### PR DESCRIPTION
**Pull request by NingLin-P**
###  What have you changed?

Currently config struct of each module on tikv are separated seating at some config.rs files that under nested directory  or inside src/config.rs, which make it hard to find a config struct of some module and also make src/config.rs large. This pr aim to solve this by moving rocksdb related config and read pool config out of src/config.rs and also moving all config.rs files to src/config directory (i.e src/server/config.rs -> src/config/server.rs).

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- Manual test
``` 
cargo check 
make clippy
```
###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

File: src/config/raftstore.rs [hunk 1]
```diff
@@ -136,7 +136,8 @@ pub struct Config {
 
 impl Default for Config {
     fn default() -> Config {
-        let split_size = ReadableSize::mb(coprocessor::config::SPLIT_SIZE_MB);
+        // let split_size = ReadableSize::mb(coprocessor::config::SPLIT_SIZE_MB);
```
Comment (overvenus) [inline src/config/raftstore.rs]: Please remove this line.

**Comment by siddontang** [comment]
I don't know whether this is a good idea, but it seems that nearly now all the modules depend on Config and the Config knows the above modules too, a little circular dependency.

What do you think? @BusyJay @nrc

**Comment by BusyJay** [comment]
I don't think it's a good idea. All configuration should be in its own module. This change prevents extracting modules as components.

**Comment by NingLin-P** [comment]
> I don't think it's a good idea. All configuration should be in its own module. This change prevents extracting modules as components.

I think extracting modules as components will just like before i.e moving code and change import path.

**Comment by nrc** [comment]
>  seems that nearly now all the modules depend on Config and the Config knows the above modules too, a little circular dependency.

It looks to me like the modules depend on config, but config does not know about any of the modules. So the dependency graph here is a tree, not circular.

I think there is a benefit to having all the config handling in one place - it means we don't have similar code scattered around the whole tree (contributing to making each module look big and complex). I hope that by putting all the config together, there will be opportunity to reduce code dup.

I also think it is nice if the config code matches the user-facing config more closely than the implementation, but that is a trade-off against keeping config local.

**Comment by BusyJay** [comment]
> ... graph here is a tree, not circular.
> I think extracting modules as components will just like before..

@nrc @NingLin-P you are right.

> ...have similar code scattered around...

Can you give some examples that will be reduced when unify them in one place?

> ...config code matches the user-facing config more closely...

I think code is about implement not the way how users use. For latter purpose, users should read the documents or configuration template instead.

In my opinion, we should put configurations definitions and validation close to the module implements, as the module knows why each configuration entry is defined and what's correct value. It's just meaningless to construct an arbitrary configuration.

Anyway, whether it's close to the module doesn't block this PR. The configuration file can still be moved to the module when extracted.

**Pull request closed without merging**